.github/workflows/infrastructure.yaml

```diff
@@ -1,4 +1,4 @@
-name: Infrastructure
+name: Deploy Terraform Infrastructure
```

Comment:
It's not always a deployment 😛 On PRs you only get the plan, so I prefer personally to keep a short and generic name (like "Infrastructure"). 

Short is also good because it doesn't clutter as much the UI (Github Actions reusable workflows names are concatenated via a `/`)